packages/common/driver-definitions/src/storage.ts

```diff
@@ -385,6 +385,32 @@ export interface IDocumentServiceFactory {
 	): Promise<IDocumentService>;
 }
 
+export interface ISocketStorageDiscoveryParts {
```

Comment:
Alternates to consider - Do we need this interface, or just pass the `ISocketStorageDiscovery` around directly?  If there is a benefit to this scoped interface, maybe it should be defined as a `Pick<ISocketStorageDiscovery, ...>` to codify the relationship?